scripts/multipart_upload_via_API.py

```diff
@@ -0,0 +1,142 @@
+import os
+import hashlib
+from base64 import b64encode
+import argparse
+from hashlib import md5
+
+import requests
+import multihash
+
+
+def b64_md5(data):
+    '''Return the Base64 encoded MD5 digest of the given data
+
+    Args:
+        data: string
+            input data
+
+    Returns:
+        string - base64 encoded MD5 digest of data
+    '''
+    return b64encode(md5(data).digest()).decode('utf-8')
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description="Utility script for uploading large asset files as a multipart upload " \
+            "using the STAC API. When calling this script, summon can be used to inject secrets " \
+                "from gopass into the environment variables. To do this, call this script like " \
+                    "this: summon -p gopass -e dev|int|prod python3 multipart_upload_via_API.py " \
+                        "localhost|DEV|INT|PROD collection-name item-name assset-name " \
+                            "asset-file-path --part-size."
+    )
+    parser.add_argument(
+        "env",
+        choices=['localhost', 'DEV', 'INT', 'PROD'],
+        help="Environment to be used: localhost, DEV, INT or PROD"
+    )
+    parser.add_argument("collection", help="Name of the asset's collection")
+    parser.add_argument("item", help="Name of the asset's item")
+    parser.add_argument("asset", help="Name of the asset to be uploaded")
+    parser.add_argument("filepath", help="Local path of the asset file to be uploaded")
+    parser.add_argument(
+        "--part-size",
+        type=int,
+        default=250,
+        help="Size of the file parts in MB [Integer, default: 250 MB]"
+    )
+
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+
+    # parsing the args and defining variables
+    args = get_args()
+
+    scheme = 'https'
+
+    if args.env == "localhost":
+        hostname = "http://127.0.0.1:8000"
+    elif args.env == "DEV":
+        hostname = "sys-data.dev.bgdi.ch"
+    elif args.env == "INT":
+        hostname = "sys-data.int.bgdi.ch"
+    else:
+        hostname = "data.geo.admin.ch"
+
+    collection = args.collection
+    item = args.item
+    asset = args.asset
+    asset_path = f'collections/{collection}/items/{item}/assets/{asset}'
+    user = os.environ.get('STAC_USER', 'unknown-user')
+    password = os.environ.get('STAC_PASSWORD', 'unknown-password')
```

Comment:
Here if no credential is provided either simply let the script crash with `KeyError` or catch this error and stop the script with a correct message. Otherwise you will hide the reason why the script is not working, I'm pretty sure those unknown credential will never be valid.

Reply:
Good idea, I'll adapt the script here